frontend/src/modules/frameworkImport/views/frameworkImportView.js

```diff
@@ -65,12 +67,112 @@ define(function(require){
+      this.$('#import_upload').addClass('display-none');
+      this.sidebarView.$('.framework-import-sidebar-save-button').removeClass('show-details');
+      var $details = this.$('#import_details');
+      var $framework_versions = $details.find('.framework-versions');
+
+      if (_.isEmpty(data.pluginVersions.red)) {
+        $('.framework-import-sidebar-save-button').addClass('save');
+      } else {
+        $('.framework-import-sidebar-save-button').remove();
+      }
+
+      // Framework versions panel
+      if (data.frameworkVersions.imported !== data.frameworkVersions.installed) {
+          $framework_versions.html(Origin.l10n.t('app.importframeworkversions', {
+              importVersion: data.frameworkVersions.imported,
+              installedVersion: data.frameworkVersions.installed
+          }));
+          $framework_versions.removeClass('display-none');
+      }
+
+      // Can/Cannot be imported summary
+      this.displaySummary(data);
+
+      // Plugin list
+      this.displayPluginList(data);
+
+      $details.removeClass('display-none');
+    },
+
+    displaySummary: function(data) {
+      var $details = this.$('#import_details');
+      var $summary_title = $details.find('.import-summary .title');
+      var $summary_description = $details.find('.import-summary .description');
+      this.sidebarView.resetButtons();
+
+      if (!_.isEmpty(data.pluginVersions.red)) {
+        $summary_title.addClass('red').text(Origin.l10n.t('app.coursecannotbeimported'));
+        $summary_description.text(Origin.l10n.t('app.coursecannotbeimporteddesc'));
+        return;
+      }
+
+      $summary_title.text(Origin.l10n.t('app.coursecanbeimported'));
+
+      var greenExists = !(_.isEmpty(data.pluginVersions['green-install']) && _.isEmpty(data.pluginVersions['green-update']));
+      if (!_.isEmpty(data.pluginVersions.amber) || greenExists) {
+        $summary_title.addClass('amber');
+        $summary_description.text(Origin.l10n.t('app.coursecanbeimporteddesc'));
+        if (greenExists && _.isEmpty(data.pluginVersions.amber)) {
+          $summary_title.removeClass('amber').addClass('green');
+        }
+        return;
+      }
+
+      $summary_description.text(Origin.l10n.t('app.coursecanbeimportedwhitedesc'));
+      return;
+    },
+
+    displayPluginList: function(data) {
+      var $plugin_list = this.$('#import_details').find('.plugin-list');
+      var categoryMap = {
+        'green-install': {
+          label: 'app.plugingreeninstalllabel'
+        },
+        'green-update': {
+          label: 'app.plugingreenupdatelabel'
+        },
+        'amber': {
+          label: 'app.pluginamberlabel'
+        },
+        'red': {
+          label: 'app.pluginredlabel'
+        }
+      };
+      var categories = ['red', 'amber', 'green-update', 'green-install'];
+
+      $plugin_list.append(new FrameworkImportPluginHeadingView().$el);
+
+      for (i = 0; i < categories.length; i++) {
```

Comment:
```suggestion
      for (var i = 0, j = categories.length; i < j; i++) {
```

Reply:
I'm not sure what the benefit of this is? It's not something I've come across before!

Reply:
They're not biggies – first amend was from the linter complaining that `i` hasn't been defined.

The second amend I just slipped in at the same time for good practice – caching the length so it's not looked up with every iteration. Example [here](http://allthingsjavascript.com/blog/index.php/2016/11/16/caching-the-length-property-in-a-for-loop).

Reply:
:+1: That is a good point I hadn't thought of before, thanks